### PR TITLE
Improve output directory

### LINKED ISSUE #101: Output CSS files to a single directory instead
Right now the generated CSS files pollute the project directory. It'll be nicer to have them in a single directory e.g. `.linaria-cache` or something

### PR DESCRIPTION
**Pull request by zamotany**
Closes #101

### CODE REVIEW

File: src/babel/preval-extract/__tests__/extractStyles.spec.js [hunk 1]
```diff
@@ -142,44 +207,45 @@ describe('preval-extract/extractStyles module', () => {
 
       expect(fs.writeFileSync).toHaveBeenCalledTimes(1);
       expect(fs.writeFileSync.mock.calls[0][0]).toEqual(
-        path.join(process.cwd(), 'A.css')
+        path.join(process.cwd(), '.linaria-cache/filename.css')
```
Comment (thymikee) [inline src/babel/preval-extract/__tests__/extractStyles.spec.js]: Wonder if we should display a warning, if the `filename` is used without `single`?

File: src/babel/preval-extract/__tests__/extractStyles.spec.js [hunk 2]
```diff
@@ -142,44 +207,45 @@ describe('preval-extract/extractStyles module', () => {
...
-        path.join(process.cwd(), 'A.css')
+        path.join(process.cwd(), '.linaria-cache/A.css')
       );
       expect(fs.writeFileSync.mock.calls[1][1]).toMatch(
         '.classname{color: #ffffff}\n.other{color: #000000}'
       );
     });
 
-    it('should overwrite if the file has changed', () => {
+    it('should overwrite if the tyles has changed', () => {
```
Comment (thymikee) [inline src/babel/preval-extract/__tests__/extractStyles.spec.js]: nit: styles

File: src/babel/preval-extract/utils.js [hunk 3]
```diff
@@ -1,8 +1,21 @@
...
+  return nodePath.scope.getBinding(nodePath.node.name);
+}
+
+export function relativeToCwd(filename: string): string {
+  return path.isAbsolute(filename)
+    ? path.relative(process.cwd(), filename)
+    : filename;
+}
+
+export function makeAbsolute(filename: string): string {
```
Comment (thymikee) [inline src/babel/preval-extract/utils.js]: Correct me if I'm wrong, but shouldn't `path.resolve(filename)` solve this?

File: src/babel/preval-extract/extractStyles.js [hunk 4]
```diff
@@ -69,16 +89,26 @@ export default function extractStyles(
...
+  // Normalize current filename path.
+  const relativeCurrentFilename = relativeToCwd(currentFilename);
+  const absCurrentFilename = makeAbsolute(currentFilename);
+
+  const { single, cache, extract, outDir, filename: basename } = {
     cache: true,
     extract: true,
     single: false,
+    outDir: '.linaria-cache',
+    filename: 'styles.css',
```
Comment (thymikee) [inline src/babel/preval-extract/extractStyles.js]: can we shift those defaults to function arguments?
```js
export default function extractStyles(
  types: BabelTypes,
  program: NodePath<*>,
  currentFilename: string,
  options: {
    single?: boolean = false,
    filename?: string = 'styles.css',
    outDir?: string = '.linaria-cache',
    cache?: boolean = true,
    extract?: boolean = true,
  }
)
```

Comment (zamotany) [reply src/babel/preval-extract/extractStyles.js]: No we cannot.

The snippet you posted is invalid:
```js
options: {
    single?: boolean = false,
    filename?: string = 'styles.css',
    outDir?: string = '.linaria-cache',
    cache?: boolean = true,
    extract?: boolean = true,
  }
```
Invalid syntax.

And besides:
```js
function test(opts = { test: true }) { console.log(opts); }

test() // => { test: true }
test({ what: 'ever' }) // => { what: 'ever' }
```

Comment (thymikee) [reply src/babel/preval-extract/extractStyles.js]: TIL 👍
You could go with this then:  
```js
type Options = {
  single?: boolean,
  filename?: string,
  outDir?: string,
  cache?: boolean,
  extract?: boolean,
};

export default function extractStyles(
  types: BabelTypes,
  program: NodePath<*>,
  currentFilename: string,
  options: Options = {
    single: false,
    filename: 'styles.css',
    outDir: '.linaria-cache',
    cache: true,
    extract: true,
  }
)
```
But however you like ¯\_(ツ)_/¯

Comment (zamotany) [reply src/babel/preval-extract/extractStyles.js]: Still won't work.

**Pull request merged**